sys/include/net/ng_netconf.h

```diff
@@ -0,0 +1,93 @@
+ *
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ */
+
+#ifndef NET_CONF_H_
+#define NET_CONF_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   Global list of configuration options available throughout the
+ *          network stack, e.g. by netdev and netapi
+ */
+typedef enum {
+    NETCONF_OPT_CHANNEL,            /**< get/set channel as uint16_t in host
+                                         byte order */
+    NETCONF_OPT_IS_CHANNEL_CLR,     /**< check if channel is clear */
+    NETCONF_OPT_ADDRESS,            /**< get/set link layer address in host byte
+                                         order */
+    NETCONF_OPT_ADDRESS_LONG,       /**< long link layer address in host byte
+                                         order (e.g. IEEE802.15.4 EUI-64) */
+    NETCONF_OPT_ADDR_LEN,           /**< get the default address length a
+                                         network device expects as uint16_t in
+                                         host byte order */
+    NETCONF_OPT_NID,                /**< get/set the network ID (e.g. PAN ID in
+                                         IEEE 802.15.4) as uint16_t in host
+                                         byte order */
+    NETCONF_OPT_TX_POWER,           /**< get/set the output power for radio
+                                         devices in dBm as int16_t in host byte
+                                         order */
+    NETCONF_OPT_MAX_PACKET_SIZE,    /**< get/set the maximum packet size a
+                                         network module can handle as uint16_t
+                                         in host byte order */
+    NETCONF_OPT_PRELOADING,         /**< en/disable preloading or read the
+                                         current state. Preload using
+                                         *send_data*, send setting state to
+                                         *NETNETCONF_STATE_TX* */
+    NETCONF_OPT_PROMISCUOUSMODE,    /**< en/disable promiscuous mode or read
+                                         the current state */
+    NETCONF_OPT_AUTOACK,            /**< en/disable link layer auto ACKs or read
+                                         the current state */
+    NETCONF_OPT_PROTO,              /**< get/set the protocol for the layer
+                                         as type *ng_nettype_t*. */
+    NETCONF_OPT_STATE,              /**< get/set the state of network devices as
+                                         type *net_netconf_state_t* */
+    /* add more options if needed */
+} ng_netconf_opt_t;
+
+/**
+ * @brief   Binary parameter for enabling and disabling options
+ */
+typedef enum {
+    NETCONF_DISABLE = 0,            /**< disable a given option */
+    NETCONF_ENABLE = 1,             /**< enable a given option */
+} ng_netconf_enable_t;
+
+/**
+ * @brief   Option parameter to be used with *NETCONF_OPT_STATE* to set or get
+ *          the state of a network module
```

Comment:
As stated before: I think the word "module" is misleading here (Think of multi-modular (in the RIOT sense) protocol implementations as IPv6 or even 6LoWPAN (fragmentation and IPHC)). I would just use "state of a network device or protocol implementation".